config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Bundle.java

```diff
@@ -283,4 +305,33 @@ public void setLoadingMode(BundleLoadingOperation loadingMode) {
         this.loadingMode = loadingMode;
     }
 
+    /**
+     * Applies unique name space and version to the given entity name
+     * ::namespace::entityName::majorVersion.minorVersion
+     *
+     * @param entityName String
+     * @return String
+     */
+    public String applyUniqueName(final String entityName, final EntityBuilder.BundleType bundleType) {
+        StringBuilder uniqueName = new StringBuilder(UNIQUE_NAME_SEPARATOR);
+        uniqueName.append(getNamespace(bundleType));
+        uniqueName.append(UNIQUE_NAME_SEPARATOR);
+        uniqueName.append(entityName);
+
+        String version = getProjectInfo().getVersion();
+        if (StringUtils.isNotBlank(version)) {
+            uniqueName.append(UNIQUE_NAME_SEPARATOR);
+            String[] subVersions = version.split("\\.");
+            uniqueName.append(subVersions.length > 0 ? subVersions[0] : version);
+            uniqueName.append(".");
+            uniqueName.append(subVersions.length > 1 ? subVersions[1] : "0");
+        }
+
+        return uniqueName.toString();
+    }
+
+    public String getNamespace(final EntityBuilder.BundleType bundleType) {
+        return getProjectInfo().getGroupName();
```

Comment:
bundleType is not utilized, refactor the code

Reply:
its used in the subclass (AnnoatedBundle)